conda_lock/conda_lock.py

```diff
@@ -280,10 +280,13 @@ def make_lock_files(
             include_dev_dependencies=include_dev_dependencies,
         )
 
+        if not override_channels:
+            channels = lock_spec.channels
```

Comment:
you probably dont need this extra flag. I think you implicitly have this flag with `channels != ()`

Reply:
i.e., if any channels are provided to this function then you're overriding what's in the lock spec